packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:code_builder/code_builder.dart';
+import 'package:dynamite/src/builder/resolve_type.dart';
+import 'package:dynamite/src/builder/state.dart';
+import 'package:dynamite/src/helpers/dart_helpers.dart';
+import 'package:dynamite/src/helpers/dynamite.dart';
+import 'package:dynamite/src/models/open_api.dart';
+import 'package:dynamite/src/models/operation.dart';
+import 'package:dynamite/src/models/response.dart';
+import 'package:dynamite/src/type_result/type_result.dart';
+
+(String? dataType, String? dataValue, bool? dataNeedsAwait) resolveMimeTypeDecode(
+  final Response response,
+  final OpenAPI spec,
+  final State state,
+  final String identifier,
+) {
+  if (response.content != null) {
+    if (response.content!.length > 1) {
+      print('Can not work with multiple mime types right now. Using the first supported.');
+    }
+
+    for (final content in response.content!.entries) {
+      final mimeType = content.key;
+      final mediaType = content.value;
+
+      final result = resolveType(
+        spec,
+        state,
+        'identifier-$mimeType',
+        mediaType.schema!,
+      );
+
+      if (mimeType == '*/*' || mimeType == 'application/octet-stream' || mimeType.startsWith('image/')) {
+        return ('Uint8List', '_response.bodyBytes', true);
+      } else if (mimeType.startsWith('text/') || mimeType == 'application/javascript') {
+        return ('String', '_response.body', true);
+      } else if (mimeType == 'application/json') {
```

Comment:
This could be turned into a switch statement

Reply:
It has `mimeType.startsWith()` and I don't think that we could put it in a switch